src/PostgREST/SchemaCache.hs

```diff
@@ -760,19 +759,25 @@ tablesSqlQuery pgVer =
   "ORDER BY table_schema, table_name"
   where
     relIsPartition = if pgVer >= pgVersion100 then " AND not c.relispartition " else mempty
-    columnDefault
+    columnDefault -- typbasetype and typdefaultbin handles `CREATE DOMAIN .. DEFAULT val`,  attidentity/attgenerated handles generated columns, pg_get_expr gets the default of a column
       | pgVer >= pgVersion120 = [q|
           CASE
+            WHEN t.typbasetype  != 0  THEN pg_get_expr(t.typdefaultbin, 0)
             WHEN a.attidentity  = 'd' THEN format('nextval(%s)', quote_literal(seqsch.nspname || '.' || seqclass.relname))
             WHEN a.attgenerated = 's' THEN null
             ELSE pg_get_expr(ad.adbin, ad.adrelid)::text
-          END AS column_default,|]
+          END|]
       | pgVer >= pgVersion100 = [q|
           CASE
+            WHEN t.typbasetype  != 0  THEN pg_get_expr(t.typdefaultbin, 0)
             WHEN a.attidentity = 'd' THEN format('nextval(%s)', quote_literal(seqsch.nspname || '.' || seqclass.relname))
             ELSE pg_get_expr(ad.adbin, ad.adrelid)::text
-          END AS column_default,|]
-      | otherwise  = "pg_get_expr(ad.adbin, ad.adrelid)::text as column_default,"
+          END|]
+      | otherwise  = [q|
```

Comment:
Don't get why this line is not covered in codecov. It runs when using `postgrest-with-postgresql-9.6 postgrest-test-spec`. I've double checked the test fails when removing it.

Reply:
Ah, code coverage only covers pg 15:

https://github.com/PostgREST/postgrest/blob/0a1564ba5aaa8175dcb738271967944652cf4ce9/.github/workflows/ci.yaml#L46-L51

Reply:
Noted this on https://github.com/PostgREST/postgrest/issues/1719#issuecomment-1615254184